src/diagnostics.rs

```diff
@@ -42,6 +42,7 @@ impl MachineStopType for TerminationInfo {}
 pub enum NonHaltingDiagnostic {
     PoppedTrackedPointerTag(Item),
     CreatedAlloc(AllocId),
+    DestroyAlloc(AllocId)
```

Comment:
Given the others use "...ed" form of the verb, I think this should "DestroyedAlloc" or maybe "FreedAlloc".